fastapi/src/ksi_oidc_fastapi/session_manager.py

```diff
@@ -0,0 +1,189 @@
+import time
+import secrets
+from typing import Dict, Optional, Any
+from dataclasses import dataclass, asdict
+from fastapi import Request
+
+from ksi_oidc_common.tokens import Tokens
+from .oidc_client import get_oidc_client
+
+import asyncio
+
+@dataclass
+class SessionData:
+    """Data stored in a user session"""
+    tokens: Optional[Tokens] = None
+    nonce: Optional[str] = None
+    oauth_state: Optional[str] = None
+    created_at: float = None
+    last_accessed: float = None
+    
+    def __post_init__(self):
+        if self.created_at is None:
+            self.created_at = time.time()
+        if self.last_accessed is None:
+            self.last_accessed = time.time()
+
+    def to_dict(self) -> Dict[str, Any]:
+        """Convert to dictionary for serialization if needed"""
+        return asdict(self)
+
+    def is_expired(self, max_age: int = 300) -> bool:
+        """Check if session is expired (default 5 minutes)"""
+        return time.time() - self.last_accessed > max_age
+
+    def is_authenticated(self) -> bool:
+        """Check if user is authenticated (has valid tokens)"""
+        return self.tokens is not None
+
+
+class SessionManager:
+    """In-memory session manager for FastAPI OIDC authentication"""
+                
+    def __init__(self, session_timeout: int = 300):
```

Comment:
I think this session timeout is way too low and will trigger frequent authentication redirects if the user is not actively using the page. The access token lifespan already ensures that the user is in fact authenticated when performing requests, so this cookie session could be stored for much longer, even multiple days.